hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneSecurityUtil.java

```diff
@@ -63,8 +62,9 @@ private OzoneSecurityUtil() {
   }
 
   public static boolean isSecurityEnabled(ConfigurationSource conf) {
-    return conf.getBoolean(OZONE_SECURITY_ENABLED_KEY,
-        OZONE_SECURITY_ENABLED_DEFAULT);
+    // Creating a new SecurityConfig object ensures that we load the BouncyCastle JCE provider at the moment.
+    // TODO: find a better way to ensure the loading of the provider.
+    return new SecurityConfig(conf).isSecurityEnabled();
```

Comment:
@fapifta , I believe at least one SecurityConfig object will be created during SCM startup if security is enabled. I think we don't need to do this "new SecurityConfig(conf)" every time when isSecurityEnabled is called.

Reply:
Yes, this may be an overkill for sure, thank you for spotting it. Although the provider initialization has to be moved to the new hdds-crypto-default module, based on your suggestion I worked out an other approach.
I have moved the JCE provider initialization code to a static method in the SecurityConfig class, and I do the initialization once per runtime.

With that I added a call to the new static method to all the places where I have added the consturctor call earlier, as that initialization is necessary for certain tests to be able to run properly after the change in how we create the CertificateFactory [here](https://github.com/apache/ozone/pull/6802/files#diff-b9cbbe81428f1c896d951ee37dcee4fbfd3f19a7ec224628c9bd109f82463910R198-R203).